packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -991,12 +991,36 @@ class FileImage extends ImageProvider<FileImage> {
   Future<ui.Codec> _loadAsync(FileImage key, DecoderBufferCallback? decode, DecoderCallback? decodeDeprecated) async {
     assert(key == this);
 
-    final Uint8List bytes = await file.readAsBytes();
-    if (bytes.lengthInBytes == 0) {
+    final int lengthInBytes = await file.length();
+    if (lengthInBytes == 0) {
       // The file may become available later.
       PaintingBinding.instance.imageCache.evict(key);
       throw StateError('$file is empty and cannot be loaded as an image.');
     }
+    // TODO(jonahwilliams): we should have a way to construct an immutable buffer from a file handle
+    // or from a list of smaller buffers to avoid this extra copying. Large files cannot
+    // be loaded through a singular readAsBytes call on some platforms (windows).
+    final Uint8List bytes = await compute((File file) {
+      final int lengthInBytes = file.lengthSync();
+      final Uint8List result = Uint8List(lengthInBytes);
+      final RandomAccessFile raf = file.openSync();
+      // Same chunk size used by file.openRead.
+      const int kChunkSize = 64 * 1024;
```

Comment:
This is different than the behavior of `readAsBytes()`, and so may have different performance.